### PR TITLE
Move BuildParams class to 'minimumRuntime' source set 

### LINKED ISSUE #49766: [CI] elasticsearch-hadoop build failure
This has been failing since the 25th of November since another build problem was fixed with https://github.com/elastic/elasticsearch-hadoop/pull/1394 

```
Execution failed for task ':elasticsearch-hadoop-hive:esCluster#start'.
13:34:35 > org/elasticsearch/gradle/info/BuildParams has been compiled by a more recent version of the Java Runtime (class file version 54.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```

https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch-hadoop+7.x+periodic/181/console

### PR DESCRIPTION
**Pull request by mark-vieira**
This commit moves BuildParams and dependent classes to the 'minimumRuntime' source set in order to maintain backwards compatibility with build-tools for builds using a Java 8 runtime.

Closes #49766

### CODE REVIEW

**Comment by mark-vieira** [comment]
FYI, this is being done in `master` and being backported to `7.x` just to keep both branches as close as possible, even though #49766 only affects the `7.x` branch of the elasticsearch-hadoop project. Doing so will simply make backports from `master` to `7.x` simpler going forward, until such a time as we can remove the need for the `minimumRuntime` source set from both branches entirely.

**Comment by mark-vieira** [comment]
@elasticmachine run elasticsearch-ci/1

File: buildSrc/src/minimumRuntime/java/org/elasticsearch/gradle/info/BuildParams.java [hunk 1]
```diff
@@ -155,6 +154,7 @@ public void reset() {
                 .filter(f -> Modifier.isStatic(f.getModifiers()))
                 .forEach(f -> {
                     try {
+                        f.setAccessible(true);
```
Comment (rjernst) [inline buildSrc/src/minimumRuntime/java/org/elasticsearch/gradle/info/BuildParams.java]: can you please add a comment explaining why this is needed?

**Review by rjernst — APPROVED** [review]
LGTM

File: buildSrc/src/minimumRuntime/java/org/elasticsearch/gradle/info/BuildParams.java (hunk 1 above)
Comment (mark-vieira) [reply buildSrc/src/minimumRuntime/java/org/elasticsearch/gradle/info/BuildParams.java]: I've added a comment to explain the need for this.

**Pull request merged**